src/rust/ide/view/graph-editor/src/component/node.rs

```diff
@@ -380,6 +440,18 @@ impl NodeModel {
     pub fn visualization(&self) -> &visualization::Container {
         &self.visualization
     }
+
+    fn set_error(&self, error:&Option<Error>) {
```

Comment:
don't use `&Option<Error>`, use `Option<&Error>` instead. This is a much better design, which allows for a much better usage. Imagine that in one point in your program you have `&Error`. Then using the second API you can bpass it here without cloning. The same applies to situation where you have `&Option<Error>` - you can pass it to second api without perf overhead, while the API you proposed is mmuch more limited.